internal/armada/server/submit.go

```diff
@@ -58,33 +57,44 @@ func NewSubmitServer(
 }
 
 func (server *SubmitServer) GetQueueInfo(ctx context.Context, req *api.QueueInfoRequest) (*api.QueueInfo, error) {
-	if e := checkPermission(server.permissions, ctx, permissions.WatchAllEvents); e != nil {
-		return nil, e
+	err := checkPermission(server.permissions, ctx, permissions.WatchAllEvents)
+	var e *ErrNoPermission
+	if errors.As(err, &e) {
+		return nil, status.Errorf(codes.PermissionDenied, "[GetQueueInfo] error for queue %s: %s", req.Name, e)
+	} else if err != nil {
+		return nil, status.Errorf(codes.Unavailable, "[GetQueueInfo] error checking permissions: %s", err)
 	}
-	jobSets, e := server.jobRepository.GetQueueActiveJobSets(req.Name)
-	if e != nil {
-		return nil, e
+
+	jobSets, err := server.jobRepository.GetQueueActiveJobSets(req.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.Unavailable, "[GetQueueInfo] error getting job sets for queue %s: %s", req.Name, err)
 	}
-	return &api.QueueInfo{
+
+	result := &api.QueueInfo{
 		Name:          req.Name,
 		ActiveJobSets: jobSets,
-	}, nil
+	}
+	return result, nil
 }
 
 func (server *SubmitServer) GetQueue(ctx context.Context, req *api.QueueGetRequest) (*api.Queue, error) {
-	queue, e := server.queueRepository.GetQueue(req.Name)
-	if e == repository.ErrQueueNotFound {
-		return nil, status.Errorf(codes.NotFound, "Queue %q not found", req.Name)
-
-	} else if e != nil {
-		return nil, status.Errorf(codes.Unavailable, "Could not load queue %q: %s", req.Name, e.Error())
+	queue, err := server.queueRepository.GetQueue(req.Name)
+	var e *repository.ErrQueueNotFound
+	if errors.As(err, &e) {
+		return nil, status.Errorf(codes.NotFound, "[GetQueue] error: %s", err)
```

Comment:
I assume we've check this + other similar that the resulting error is similarly readable and useful

Reply:
This should be ok. It'll be even better in a subsequent PR.